lib/impl/updateForm.js

```diff
@@ -451,6 +458,60 @@ module.exports = function updateForm(connections, options, formToUpdate, cb) {
       });
     }
 
+    function checkRulesWithRepeatingSections() {
+      var fieldSectionMapping = getSectionMap(form);
+      fieldSectionMapping = _.mapObject(fieldSectionMapping, getSectionById);
+
+      form.fieldRules.forEach(function(fieldRule) {
+        var sectionIds = _.chain(fieldRule.ruleConditionalStatements)
+          .map(function(ruleConditionalStatement) {
+            var sourceId = ruleConditionalStatement.sourceField.toString();
+            return fieldSectionMapping[sourceId];
+          })
+          .filter(function(section) {
+            return section && section.repeating;
+          })
+          .map(function(repeatingSection) {
+            return repeatingSection._id.toString();
+          })
+          .uniq()
+          .value();
+
+        if (sectionIds.length > 1) {
+          // there are more source fields in different repeating sections
+          rulesFlaggedForDeletion[fieldRule._id.toString()] = true;
+        } else if (sectionIds.length === 1) {
+          // all source fields within repeating section are in the same section
+          // -> check that all target fields are in that same section
+          var targetInDifferrentSection = _.find(fieldRule.targetField, function(targetField) {
+            var targetSection = fieldSectionMapping[targetField.toString()];
+            var targetSectionId = targetSection && targetSection._id.toString();
+            return targetSectionId !== sectionIds[0];
+          });
+          if (targetInDifferrentSection) {
+            rulesFlaggedForDeletion[fieldRule._id.toString()] = true;
+          }
+        }
+      });
+
+      function getSectionById(id) {
+        var section;
+
+        _.find(form.pages, function(page) {
+          return _.find(page.fields, function(field) {
+            var found = field._id.toString() === id;
+            if (found) {
+              section = field;
+            }
+
+            return found;
+          });
+        });
+
+        return section;
+      }
+    }
+
```

Comment:
I'd rename function to `getFieldById` since you dont seem to be checking for fieldType